docs/types/common.ts

```diff
@@ -2,6 +2,8 @@ export type PageMetadata = {
   title: string;
   sourceCodeUrl?: string;
   maxHeadingDepth?: number;
+  /* If the page should show up in the search results, only hide when `false` */
+  searchable?: boolean;
```

Comment:
May want to consider a term like `hideFromSearch` or `excludeFromSearch` which defaults to `false` rather than `true`. I think it's a bit more self explanatory and doesn't need additional comments.

```suggestion
  hideFromSearch?: boolean;
```